### PR TITLE
Reports bar charts API client

### LINKED ISSUE #2840: Reports: Bar chart API
This is about adding the api to fetch TEs added up by time instead of project for our reports.

screen shot 2018-04-25 at 14 10 24

## Details

The relevant endpoints are:
`/reports/api/v3/mobile/workspace/{workspace_id}/summary/time_entries`
`/reports/api/v3/mobile/workspace/{workspace_id}/search/time_entries/totals`

Which should work as soon as https://github.com/toggl/backend/issues/6590 is closed.

No other documentation right now I'm afraid, but you can check how the website makes requests to their endpoints and take it from there.

### PR DESCRIPTION
**Pull request by simonrozsival**
## What's this?
This is just the implementation of the API endpoint for the bar charts.

### Relationships

Closes #2840 

## How is it done?
The deserialization is a bit tricky, because the JSON returned from the server is not really nice/easy to use for us.

There is one thing I am not sure about is the way the `"rates"` section and then the `"by_rates"` properties work. Sample server response:

```json
{
    "seconds": 4923860,
    "rates": [
        {
            "billable_seconds": 1508432,
            "hourly_rate_in_cents": 0,
            "currency": ""
        }
    ],
    "graph": [
        {
            "seconds": 67367,
            "by_rate": {
                "0": 15107
            }
        },
        {
            "seconds": 604247,
            "by_rate": {
                "0": 360678
            }
        },
        {
            "seconds": 583420,
            "by_rate": {
                "0": 354046
            }
        },
        {
            "seconds": 492846,
            "by_rate": {}
        },
        {
            "seconds": 649492,
            "by_rate": {
                "0": 181976
            }
        },
        {
            "seconds": 777571,
            "by_rate": {
                "0": 68627
            }
        },
        {
            "seconds": 687541,
            "by_rate": {
                "0": 355875
            }
        },
        {
            "seconds": 693920,
            "by_rate": {
                "0": 153146
            }
        },
        {
            "seconds": 267460,
            "by_rate": {
                "0": 8948
            }
        },
        {
            "seconds": 49739,
            "by_rate": {
                "0": 7671
            }
        },
        {
            "seconds": 50247,
            "by_rate": {
                "0": 2349
            }
        },
        {
            "seconds": 10,
            "by_rate": {
                "0": 9
            }
        }
    ],
    "resolution": "month"
}
```

### Side effects
No side effects.

## Review hints
Check the integration tests: https://app.bitrise.io/build/9b0c94a3860f020b (2 random tests are failing, but none of them is related to this PR)

## :squid: Permissions
I will do it myself this time.

### CODE REVIEW

**Comment by simonrozsival** [comment]
Sample response for multiple billable rates (workspace, project):
```json
{
    "seconds": 1500,
    "rates": [
        {
            "billable_seconds": 800,
            "hourly_rate_in_cents": 2500,
            "currency": "USD"
        },
        {
            "billable_seconds": 300,
            "hourly_rate_in_cents": 5000,
            "currency": "USD"
        }
    ],
    "graph": [
        {
            "seconds": 0,
            "by_rate": {}
        },
        {
            "seconds": 1500,
            "by_rate": {
                "0": 800,
                "1": 300
            }
        }
    ],
    "resolution": "day"
}
```

**Comment by amulware** [comment]
We need to clarify the `by_rate` thing to make sure we display nonbillable and billable correctly. Right now, it looks like maybe you just sum those up and that's your billable part, compared to the `seconds` total?

**Comment by simonrozsival** [comment]
Yes, right now I sum all the values of `"by_rate"` which seems to be the billable seconds according to different billable rates. Since we don't care how much money exactly it represents, it seems natural to just sum it up and I think this is the best thing we can do with this kind of data.

I didn't check Notion for any topic on billable rates, I just read the support page (https://support.toggl.com/billable-rates/).

File: Toggl.Ultrawave.Tests.Integration/TimeEntriesReportsApiTests.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            totals.Groups[0].Billable.Should().Be(TimeSpan.FromSeconds(2));
+            totals.Groups[1].Total.Should().Be(TimeSpan.FromSeconds(7));
+            totals.Groups[1].Billable.Should().Be(TimeSpan.FromSeconds(4));
+        }
+
+        [Theory, LogTestInfo]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(33)]
+        [InlineData(34)]
```
[the remark this answers is not in the document]
Comment (simonrozsival) [reply Toggl.Ultrawave.Tests.Integration/TimeEntriesReportsApiTests.cs]: Check https://github.com/toggl/mobileapp/issues/2839

**Pull request merged**